### ISSUE TITLE
k_mem_pool crash with larger values of n_max

### ISSUE THREAD

**Issue by andrewboie** [zephyrproject-rtos/zephyr#6727]
Steps to reproduce:

At toplevel declare:
`K_MEM_POOL_DEFINE(my_pool, 64, 1024, 61, 4);`

Where n_max is 61.

Then somewhere in code:
```
	struct k_mem_block blk;

	mem = k_mem_pool_alloc(&my_pool, &blk, 1024, 0);
```

This results in a page fault on x86, on further inspection, the dlist structures in pool->levels are being corrupted:

***** CPU Page Fault (error code 0x00000000)
Supervisor thread read address 0x20000003

If this is not a bug, we need to clearly document what the upper bounds for various parameters to K_MEM_POOL_DEFINE are, and add assertions to enforce them.

**Comment by andyross** [comment]
Yeah, there's nothing wrong or out of bounds with that code.  This should work, it's a real bug.  Will investigate.

**Comment by linkmeyer** [comment]
@andyross , what's the status?

**Comment by andyross** [comment]
Last I tried this, I couldn't reproduce it.  But it needs investigation.  Will be fixed or closed for 1.13.
